src/java.base/unix/classes/java/lang/ProcessEnvironment.java

```diff
@@ -63,6 +66,15 @@
     private static final HashMap<Variable,Value> theEnvironment;
     private static final Map<String,String> theUnmodifiableEnvironment;
     static final int MIN_NAME_LENGTH = 0;
+    private static final Charset jnuCharset;
+    static {
+        @SuppressWarnings("removal")
+        String jnuEncoding = AccessController.doPrivileged((PrivilegedAction<String>) ()
+            -> System.getProperty("sun.jnu.encoding"));
```

Comment:
Hello @naotoj .
If I just use `System.getProperty("sun.jnu.encoding")`, following testcases were failed.
```
java/lang/ProcessBuilder/SecurityManagerClinit.java 
java/lang/ProcessHandle/PermissionTest.java 
```
Please give me your suggestion again.

Reply:
Ah, OK. Never mind.

Reply:
It might be worthwhile to cache the `sun.jnu.encoding` property value in jdk/internal/util/StaticProperty.
And perhaps even cache the Charset (not just the string).